app/assets/javascripts/download_util.js

```diff
@@ -0,0 +1,53 @@
+(function () {
+  OSM.downloadBlob = function (blob, filename) {
+    const url = URL.createObjectURL(blob);
+    const a = document.createElement("a");
+    a.href = url;
+    a.download = filename;
+    document.body.appendChild(a);
+    a.click();
+    document.body.removeChild(a);
+    URL.revokeObjectURL(url);
+  };
+
+  class DownloadUtil {
+    static async handleExportSuccess(fetchResponse, downloadCallback) {
+      try {
+        const blob = await fetchResponse.response.blob();
+        downloadCallback(blob);
+      } catch (err) {
+        // eslint-disable-next-line no-alert
+        alert(OSM.i18n.t("javascripts.share.export_failed", { reason: "(blob error)" }));
+      }
+    }
+
+    static async handleExportError(event) {
+      let detailMessage;
+      try {
+        detailMessage = event?.detail?.error?.message;
+        if (!detailMessage) {
+          const responseText = await event.detail.fetchResponse.responseText;
+          const parser = new DOMParser();
+          const doc = parser.parseFromString(responseText, "text/html");
+          detailMessage = doc.body ? doc.body.textContent.trim() : "(unknown)";
+        }
+      } catch (err) {
+        detailMessage = "(unknown)";
+      }
+      // eslint-disable-next-line no-alert
+      alert(OSM.i18n.t("javascripts.share.export_failed", { reason: detailMessage }));
+    }
+
+    static getTurboBlobHandler(downloadCallback) {
```

Comment:
This is setup to return a function that captures the callback but in fact both users are using very similar callbacks - in particular they are both just forwarding to `OSM.downloadBlob`.

So wouldn't make more sense for this just to take the filename (which is the only thing that changes and needs to be captured) and put the `downloadBlob` call in here?

Reply:
Indeed, and once all that is done `downloadBlob` doesn't really need to be exported in `OSM` and could just be a function in the `DownloadUtil` class?